### PR TITLE
Add s3stalker.py

### PR DESCRIPTION
**Pull request by mraspaud**
This adds the s3stalker to poll an s3 bucket. Example usage:
```shell
python bin/s3stalker.py sentinel-s3-ol2wfr-zips/2020/12/08 examples/s3stalker.yaml
```

### CODE REVIEW

File: pytroll_collectors/s3stalker.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                matching_files.append(file)
+            except ValueError:
+                pass
+        return matching_files
+    return files
+
+
+def set_last_fetch(timestamp):
+    """Set the last fetch time."""
+    global last_fetch
```
Comment (pnuu) [inline pytroll_collectors/s3stalker.py:75]: If the s2stalker were a class the `global` statement wouldn't be needed

File: bin/s3stalker.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    LOGGER = logging.getLogger('s3stalker')
+
+    time_back = config['timedelta']
+    subject = config['subject']
+    pattern = config['file_pattern']
+
+    from posttroll.publisher import Publish
+    try:
+        with Publish("s3_stalker") as pub:
```
Comment (pnuu) [inline bin/s3stalker.py:80]: I think this branch should be in the `pytroll_collectors.s2stalker` module.

**Review by pnuu — COMMENTED** [review]
I'm not familiar with S3, but few comments inline.

File: bin/s3stalker.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+from dateutil import tz
+
+from pytroll_collectors import s3stalker
+from pytroll_collectors.helper_functions import read_yaml
+
+LOGGER = logging.getLogger(__name__)
+
+
+@contextmanager
+def sleeper(duration):
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply bin/s3stalker.py:40]: Yes, see here for example: https://book.pythontips.com/en/latest/context_managers.html#implementing-a-context-manager-as-a-generator

File: bin/s3stalker.py (hunk 2 above)
Comment (mraspaud) [reply bin/s3stalker.py]: Fixed

File: pytroll_collectors/s3stalker.py (hunk 1 above)
Comment (mraspaud) [reply pytroll_collectors/s3stalker.py]: Fixed

File: pytroll_collectors/s3stalker.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+import datetime
+import json
+
+import fsspec.implementations.zip
+import s3fs
+from dateutil import tz
+from posttroll.message import Message
+from trollsift import Parser
+import posixpath
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply pytroll_collectors/s3stalker.py]: Fixed

File: bin/s3stalker.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    from posttroll.publisher import Publish
+    try:
+        with Publish("s3_stalker") as pub:
+            with sleeper(2.5):
+                s3stalker.set_last_fetch(datetime.now(tz.UTC) - timedelta(**time_back))
+                s3_kwargs = config['s3_kwargs']
+                fs, files = s3stalker.get_last_files(bucket, pattern=pattern, **s3_kwargs)
+                messages = s3stalker.filelist_unzip_to_messages(fs, files, subject)
+
+            for message in messages:
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply bin/s3stalker.py]: Yes, I now added an explanation in the script main docstring.

File: bin/s3stalker.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            log_dict = yaml.safe_load(fd.read())
+            logging.config.dictConfig(log_dict)
+
+    LOGGER = logging.getLogger('s3stalker')
+
+    time_back = config['timedelta']
+    subject = config['subject']
+    pattern = config['file_pattern']
+
+    from posttroll.publisher import Publish
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply bin/s3stalker.py]: Fixed

File: bin/s3stalker.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    parser.add_argument("bucket", help="The bucket to retrieve from.")
+    parser.add_argument("config", help="Config file to be used")
+    parser.add_argument("-l", "--log",
+                        help="Log configuration file",
+                        default=None)
+
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply bin/s3stalker.py]: Fixed

**Comment by mraspaud** [comment]
@pnuu ready for re-review if you want, merge otherwise.

File: pytroll_collectors/s3stalker.py [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Module to file new files on an s3 bucket."""
```
Comment (pnuu) [inline pytroll_collectors/s3stalker.py:22]:
```suggestion
"""Module to fine new files on an s3 bucket."""
```

**Review by pnuu — APPROVED** [review]
LGTM, fixed one typo in a docstring.

**Comment by mraspaud** [comment]
Thanks!

**Pull request merged**